internal/server/v2/observation/calculation_util.go

```diff
@@ -245,3 +246,34 @@ func evalExpr(
 		return nil, fmt.Errorf("unsupported ast type %T", t)
 	}
 }
+
+func formatCalculatedResponse(
+	resp *pbv2.ObservationResponse,
+	equation *Equation,
+) error {
+	// Replace placeholder by final variable.
+	variableObs, ok := resp.ByVariable[INTERMEDIATE_NODE]
+	if !ok {
+		return fmt.Errorf("missing intermediate variable in intermediate response")
+	}
+	resp.ByVariable[equation.variable] = variableObs
+	delete(resp.ByVariable, INTERMEDIATE_NODE)
+
+	// Update Facets with IsDcAggregate=true.
+	newFacets := map[string]*pb.Facet{}
+	facetIdMap := map[string]string{}
+	for oldFacetId, oldFacet := range resp.Facets {
+		newFacet := oldFacet
```

Comment:
Doing this will update `oldFacet` as well. Wondering if you should first clone it and then update just the clone.

```
newFacet := proto.Clone(oldFacet).(*pb.Facet)
```

Reply:
Good catch, thanks! Updated